crates/common/flockfile/src/unix.rs

```diff
@@ -44,24 +53,22 @@ impl Flockfile {
     ///
     pub fn new_lock(path: impl AsRef<Path>) -> Result<Flockfile, FlockfileError> {
         let path = PathBuf::new().join(path);
-        let file = match OpenOptions::new()
+        let file = OpenOptions::new()
             .create(true)
             .read(true)
             .write(true)
             .open(&path)
-        {
-            Ok(file) => file,
-            Err(err) => {
-                return Err(FlockfileError::FromIo { path, source: err });
-            }
-        };
-
-        match flock(file.as_raw_fd(), FlockArg::LockExclusiveNonblock) {
-            Ok(()) => (),
-            Err(err) => {
-                return Err(FlockfileError::FromNix { path, source: err });
+            .map_err(|source| FlockfileError::FromIo {
+                path: path.clone(),
+                source,
+            })?;
```

Comment:
Nitpicking. Just to be consistent with the following `map_err` and highlight that the `source` is actually an error.

```suggestion
            .map_err(|err| FlockfileError::FromIo {
                path: path.clone(),
                source: err,
            })?;
```

Reply:
Fixed in the fixup commit (will squash once approved).

Btw: You are really welcome to use the use the "Request changes" review option in cases like this and block merging with it. Even (or maybe _especially_) when it is nitpicking!